ibis/expr/types/generic.py

```diff
@@ -1318,9 +1319,69 @@ def __array__(self, dtype=None):
         return self.execute().__array__(dtype)
 
     def __interactive_rich_console__(self, console, options):
+        return console.render(self.preview(), options=options)
+
+    def preview(
+        self,
+        *,
+        max_rows: int | None = None,
+        show_types: bool | None = None,
+        max_length: int | None = None,
+        max_string: int | None = None,
+        max_depth: int | None = None,
+        console_width: int | float | None = None,
+    ) -> RichTable:
+        """Pretty-print a preview using rich.
+
+        This is an explicit version of what you get when you inspect
+        this object in interactive mode, except with this version you
+        can pass formatting options. The options are the same as those exposed
+        in `ibis.options.interactive`.
+
+        Parameters
+        ----------
+        max_rows
+            Maximum number of rows to display
+        show_types
```

Comment:
Can you remove this argument? I'm not sure there's a reason to hide type information.

Reply:
roger, I will do this. Still waiting to hear back on any other feedback (especially my question of returning vs printing the rich table) before I go in an change everything at the same time.

Reply:
done.